lib/surface/compiler.ex

```diff
@@ -719,6 +719,14 @@ defmodule Surface.Compiler do
 
       for prop_name <- missing_props_names do
         message = "Missing required property \"#{prop_name}\" for component <#{meta.node_alias}>"
+
+        message =
+          if prop_name == :id do
```

Comment:
we do some stuff that allows using an `id` prop within a `Surface.Component` without it turning into a stateful component, and so we need to check if the module is stateful or stateless; which you can do with the `is_stateful_component/1` function:

```
if prop_name == :id and is_stateful_component(meta.module) do
  ...
else
  ...
end
```

Could you also update the error message to something like:

```elixir 
"""
Missing required property "#{prop_name}" for component <#{meta.node_alias}>

Hint: Components using Surface.LiveComponent automatically define a required `id` component to make them stateful. If you meant to create a stateless component, you can switch to `use Surface.Component`.
"""
```

Reply:
you got it!

Reply:
@leandrocp updated!